Packs/HackerOne/Integrations/HackerOne/HackerOne.py

```diff
@@ -0,0 +1,588 @@
+
+        :type response: ``requests.models.Response``
+        :param response: response from API.
+
+        :raises: raise DemistoException based on status code of response.
+        """
+
+        if response.headers.get("Content-Type") and ("text/html" in response.headers["Content-Type"]):
+            raise DemistoException(MESSAGES['COMMON_ERROR_MESSAGE'])
+
+        err_msg = None
+        if response.status_code == 401:
+            err_msg = HTTP_ERROR[401]
+
+        elif response.status_code >= 500:
+            err_msg = HTTP_ERROR[500]
+
+        elif response.status_code == 404:
+            err_msg = HTTP_ERROR[404]
+
+        elif response.status_code == 403:
+            err_msg = HTTP_ERROR[403]
+
+        else:
+
+            # Parse json error response
+            errors = response.json().get("errors", [])
+            if not errors:
+                raise DemistoException(MESSAGES['COMMON_ERROR_MESSAGE'])
+
+            for error in errors:
+                msg = error.get("detail", error.get("title", MESSAGES['COMMON_ERROR_MESSAGE']))
+
+                if err_msg:
+                    err_msg = f"{err_msg}\n{msg}"
+                else:
+                    err_msg = msg
+
+        raise DemistoException(err_msg)
+
+
+''' HELPER FUNCTIONS '''
+
+
+def prepare_filter_by_arguments(program_handle, severity, state, filters) -> Dict[str, Any]:
+    """
+    Prepares params for the filters provided by user
+
+    :type program_handle: ``List``
+    :param program_handle: The program handle provided by the user.
+
+    :type severity: ``Any``
+    :param severity: Severity level provided by user.
+
+    :type state: ``Any``
+    :param state: State provided by user.
+
+    :type filters: ``str``
+    :param filters: The advanced_filter argument provided by the user.
+
+    :return: Parameters related to the filters.
+    :rtype: ``Dict[str, Any]``
+    """
+    params = {"filter[program][]": program_handle, 'filter[severity][]': severity,
+              'filter[state][]': state}
+
+    if not filters:
+        return params
+
+    filters = json.loads(filters)
+    for key, value in filters.items():
+        key, value = key.strip(), value.strip()
+        if not key or not value:
+            continue
+
+        if "[]" in key:
+            params[key] = argToList(value)
+        else:
+            params[key] = value
+
+    return params
+
+
+def validate_fetch_incidents_parameters(max_fetch: Optional[int], program_handle: List, filters: str):
+    """
+    Validates fetch incident parameters, raise ValueError on invalid arguments.
+
+    :type max_fetch: ``int``
+    :param max_fetch: Maximum number of incidents per fetch provided by user.
+
+    :type program_handle: ``List``
+    :param program_handle: The program handle provided by the user.
+
+    :type filters: ``str``
+    :param filters: The advanced_filter argument provided by the user.
+
+    """
+    if not 0 < max_fetch <= 100:  # type:ignore
```

Comment:
why do we fail if max_fetch>100? isn't this the same validation as the common args validation?

Reply:
@ShahafBenYakir API of HackerOne has a limit of pulling 100 reports. So we have kept a validation and although the validation is the same as the common args validation, an error message is different.

Reply:
Let's go over this on tomorrow's call

Reply:
Can we keep max_fetch limit as 100 only, as at max we can support 200 and since API has validation of max fetch limit as 100 we will have to do 2 API calls if we allow user to add any value upto 200. This additional API call for fetch incident just to support limit of 200 is adding a complexity to recommended standards of implementation.

Reply:
PMd you on slack